src/java/src/main/native/ai_onnxruntime_genai_Generator.cpp

```diff
@@ -78,3 +78,21 @@ Java_ai_onnxruntime_genai_Generator_getSequenceLastToken(JNIEnv* env, jobject th
 
   return jint(tokens[num_tokens - 1]);
 }
+
+JNIEXPORT void JNICALL
+Java_ai_onnxruntime_genai_Generator_setActiveAdapter(JNIEnv* env, jobject thiz, jlong native_handle,
+                                                     jlong adapters_native_handle, jstring adapter_name) {
+  CString name{env, adapter_name};
+  ThrowIfError(env, OgaSetActiveAdapter(reinterpret_cast<OgaGenerator*>(native_handle),
+                                        reinterpret_cast<OgaAdapters*>(adapters_native_handle),
+                                        name));
+}
+
+JNIEXPORT jlong JNICALL
+Java_ai_onnxruntime_genai_Generator_getOutputNative(JNIEnv* env, jobject thiz, jlong native_handle,
+                                                    jstring output_name) {
+  OgaTensor* tensor = nullptr;
+  CString name{env, output_name};
+  ThrowIfError(env, OgaGenerator_GetOutput(reinterpret_cast<OgaGenerator*>(native_handle), name, &tensor));
+  return reinterpret_cast<jlong>(tensor);
```

Comment:
IIRC you need to explicitly check the return type from ThrowIfError and return as it's doesn't quite behave as expected. The 'throw' is queued up but does not cause the function to return.

Reply:
Should I check the return value of `ThrowIfError` everywhere? There's a lot of usage in the existing code that do not check it. What's the downside for not checking it specifically? It'll still throw the exception, right? Will the return value become garbage?

Reply:
I've added the check of return value. I think there's other places where the return value of `ThrowIfError` is not correctly checked. I'll try to fix them later.